src/commonTest/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandlerTestsCommon.kt

```diff
@@ -1149,6 +1151,38 @@ class IncomingPaymentHandlerTestsCommon : LightningTestSuite() {
         }
     }
 
+    @Test
+    fun `purge expired incoming payments`() = runSuspendTest {
+        val paymentHandler = IncomingPaymentHandler(TestConstants.Bob.nodeParams, TestConstants.Bob.walletParams, InMemoryPaymentsDb())
+
+        // create incoming payment that has expired and not been paid
+        val expiredInvoice = paymentHandler.createInvoice(randomBytes32(), defaultAmount, "expired", listOf(), expirySeconds = 3600,
+            timestampSeconds = currentTimestampSeconds() - 3600 - 60)
+
+        // create incoming payment that has expired and been paid
+        val paidInvoice = paymentHandler.createInvoice(defaultPreimage, defaultAmount, "paid", listOf(), expirySeconds = 3600,
+            timestampSeconds = currentTimestampSeconds() - 3600 + 2)
+
+        // pay incoming invoice before it expires
+        val add = makeUpdateAddHtlc(0, randomBytes32(), paymentHandler, paidInvoice.paymentHash, makeMppPayload(defaultAmount, defaultAmount, paidInvoice.paymentSecret))
+        val result = paymentHandler.process(add, TestConstants.defaultBlockHeight)
+        assertTrue { result is IncomingPaymentHandler.ProcessAddResult.Accepted }
+        delay(2_000) // wait for 2 seconds so paid invoice expires
```

Comment:
I'd like to avoid introducing delays in unit tests, otherwise that tends to creep in many places and we end up with a very slow test runs. We really want our test suite to run as fast as it can.

We can achieve by creating the received payment directly in the DB instead of going through `paymentHandler.process`:

```kotlin
val db = InMemoryPaymentsDb()
val paymentHandler = IncomingPaymentHandler(TestConstants.Bob.nodeParams, TestConstants.Bob.walletParams, db)

// create incoming payment that has expired and not been paid
val expiredInvoice = paymentHandler.createInvoice(randomBytes32(), defaultAmount, "expired", listOf(), expirySeconds = 3600,
    timestampSeconds = currentTimestampSeconds() - 3600 - 60)

// create incoming payment that has expired, but has been paid before its expiry
val paidInvoice = paymentHandler.createInvoice(defaultPreimage, defaultAmount, "paid", listOf(), expirySeconds = 3600,
    timestampSeconds = currentTimestampSeconds() - 3600 - 60)
db.receivePayment(defaultPaymentHash, setOf(IncomingPayment.ReceivedWith.LightningPayment(defaultAmount, randomBytes32(), 3)), receivedAt = currentTimestampSeconds() - 1800)
```

Reply:
Nice! It's easier to read this way too. Fixed in [abd5760](https://github.com/ACINQ/lightning-kmp/pull/322/commits/abd576080cd4b240c509346e83efbb90ac2b6a71).